packages/api/src/server/analytics/service/analytics.factory.ts

```diff
@@ -78,4 +111,25 @@ export class AnalyticsFactory {
       { $sort: { env: 1 } }
     ];
   }
+
+  async getStartAndEndDate(days: number): Promise<Date[]> {
+    const endDate = new Date();
+    const startDate = new Date();
+    startDate.setDate(startDate.getDate() - days);
+    return [startDate, endDate];
+  }
+
+  groupEphemeral(response: Object, obj: any) {
+    const result = response[AnalyticsFactory.ephemeral].findIndex((element) => element.startDate === obj.startDate);
+    if (result !== -1) response[AnalyticsFactory.ephemeral][result].count += obj.count;
+    else {
+      obj.env = AnalyticsFactory.ephemeral;
+      response[AnalyticsFactory.ephemeral].push(obj);
+    }
```

Comment:
```suggestion
    if (result === -1) {
    obj.env = AnalyticsFactory.ephemeral;
      response[AnalyticsFactory.ephemeral].push(obj);
      return;
    }
    response[AnalyticsFactory.ephemeral][result].count += obj.count;
```
can we do something like this?